File: banchus-common/src/main/java/pe/nanamochi/banchus/domain/dto/Geolocation.java
package pe.nanamochi.banchus.domain.dto;

public record Geolocation(
    String status,
    String country,
    String countryCode,
    String region,
    String regionName,
    String city,
    String zip,
    float lat,
    float lon,
    String timezone,
    String isp,
    String org,
    String as,
    String query) {
  public static Geolocation local() {
    return new Geolocation(
        null, null, "kp", null, null, null, null, 0.0f, 0.0f, null, null, null, null, null);
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/dto/external/Geolocation.kt
package pe.nanamochi.banchus.dto.external

import com.fasterxml.jackson.annotation.JsonProperty

data class Geolocation(
    val status: String? = null,
    val country: String? = null,
    val countryCode: String? = null,
    val region: String? = null,
    val regionName: String? = null,
    val city: String? = null,
    val zip: String? = null,
    @JsonProperty("lat") val latitude: Float = 0.0f,
    @JsonProperty("lon") val longitude: Float = 0.0f,
    val timezone: String? = null,
    val isp: String? = null,
    val org: String? = null,
    val query: String? = null,
) {
    companion object {
        fun local() = Geolocation(status = "success", countryCode = "XX", country = "Local")
    }
}
